Deprecate legacy doc (#18585)
nrfconnect/sdk-nrf: doc/nrf/libraries/security/nrf_security/doc/backend_config.rst

This section covers the configurations available when using :ref:`legacy_crypto_support`.

.. caution::

>    Legacy crypto toolbox APIs is marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.

Reviewer: ```suggestion
   Legacy crypto toolbox APIs are marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
```

> .. caution::
>    Legacy crypto toolbox APIs is marked as deprecated in nRF Connect SDK 2.8.0, and will removed in a future version.
>    It is not recommended to rely on the configurations presented here or the APIs they configure in any new designs.

Reviewer: ```suggestion
.. caution::
   Legacy crypto toolbox APIs is marked as deprecated in the |NCS| version 2.8.0, and will be removed in a future version.
   It is not recommended to use the legacy crypto toolbox APIs and the related configurations in any new designs.
```

Reviewer 2: Subject-verb disagreement with 'APIs is'. APIs are?

Reviewer: Argh, good catch. I read API LOL
